File: person.rb
class Person
  def initialize(name, city, age)
    @name = name
    @city = city
    @age = age
  end

  def adult?
    @age >= 18
  end
end

messi = Person.new('Lionel', 'Barcelona', 24)
puts messi.adult? # true

File: Person.scala
class Person(val name: String, val city: String, val age: Int) {
  def isAdult = age >= 18
}

val messi = new Person("Lionel", "Barcelona", 24)
println(messi.isAdult) // true
println(messi.name) // Lionel

messi.name = "Leo" // compile error: reassignment to val
